File: shared/domain/translation/services/TranslationDomainService.kt
package shared.domain.translation.services

import shared.domain.translation.entities.TranslationResult
import shared.domain.translation.ports.inbound.TranslationService
import shared.domain.translation.ports.outbound.LlmGateway

/**
 * 翻译领域服务
 * Translation 上下文的核心服务实现
 */
class TranslationDomainService(
    private val llmGateway: LlmGateway
) : TranslationService {

    override suspend fun translate(text: String, targetLanguage: String): TranslationResult {
        return llmGateway.translate(
            text = text,
            sourceLanguage = "zh",
            targetLanguage = targetLanguage
        )
    }
}

File: desktop/src/services/TranslationDomainService.ts
import type { TranslationResult } from '../domain/translation/entities';
import type { TranslationService } from '../domain/translation/ports';
import type { LlmGateway } from '../domain/translation/ports';

export class TranslationDomainService implements TranslationService {
  constructor(private llmGateway: LlmGateway) {}

  async translate(
    text: string,
    targetLanguage: string
  ): Promise<TranslationResult> {
    return this.llmGateway.translate(text, 'zh', targetLanguage);
  }
}
